### DISCUSSION TITLE
How to move files out of running firejail sandbox?

### DISCUSSION THREAD

**Discussion by pjhfggij** [netblue30/firejail#6516] [Q&A] [answered]
I broke brain trying to understand [how --private works](https://github.com/netblue30/firejail/discussions/6513) and why it doesn't let --whitelist to "do what i mean", so I'm wondering instead if there is any way to move files out of firejail sandbox after it's started, something like:
```
firejail copy <PID>:/home/user/Downloads/file .  
```

sadly python is blocked by default when --private option is used so there is no way to `firejail join` and `python -m http.server`. Socat works, but it's horrible ux

I refuse to believe that you haven't come across this issue at any point. I just couldn't find anything in neither issues nor discussions

**Comment by rusty-snake** [comment, marked as answer]
> firejail copy <PID>:/home/user/Downloads/file .  

`firejail --get=PID /home/user/Downloads/file`

**Reply by pjhfggij** [reply]
fantastic, thank you, I've completely missed --get --put and --ls options
